cluster/src/main/scala/org/apache/spark/sql/SnappyContextFactory.scala

```diff
@@ -36,11 +36,12 @@ class SnappyContextFactory extends SparkContextFactory {
 
 object SnappyContextFactory {
 
-  private[this] val snappyContextLike =
-    SnappyContext(LeadImpl.getInitializingSparkContext)
+  private[this] val snappySession =
+    new SnappySession(LeadImpl.getInitializingSparkContext)
+
+  protected def newSession(): SnappySession with ContextLike =
+    new SnappySession(snappySession.sparkContext) with ContextLike {
```

Comment:
The spark state is not being shared. We need to share the state across multiple SnappySessions. I would recommend fixing the builder API if there is no quick fix for this issue.